faster_sam/middlewares/lambda_authorizer.py

```diff
@@ -35,17 +86,26 @@ class LambdaAuthorizerMiddleware(BaseHTTPMiddleware):
             Client Lambda object
     """
 
-    def __init__(self, app: ASGIApp, arn: str, client: Optional[BaseClient] = None) -> None:
+    def __init__(
+        self, app: ASGIApp, lambda_function: str, credentials: Optional[Credentials] = None
+    ) -> None:
         """
         Initializes the LambdaAuthorizer.
         """
         super().__init__(app, self.dispatch)
-        self.arn = arn
+        self.lambda_function = lambda_function
+        self.credentials = credentials
+        self._client = None
+
+    @property
+    def client(self) -> BaseClient:
+        if self._client is None:
+            self._client = Client(self.credentials)
+
+        if self._client.expired():
+            self._client = Client(self.credentials)
```

Comment:
Se for criar uma classe para descrever este cliente customizado acho que seria melhor manter a instância e chamar algum método que faça só o necessário para renovar a sessão ao invés de recriar tudo de novo, se for seguir nessa linha seria mais simples criar apenas uma função.

Eu criaria um método chamado `refresh` ou `refresh_credentials` e invocaria ele aqui.